### PR TITLE
Correct case of `useMultistageEngine` property reference in  `JsonAsyncHttpPinotClientTransportFactory`

### LINKED ISSUE #11819: JsonAsyncHttpPinotClientTransportFactory has wrong case for `useMultiStageEngine` property
`JsonAsyncHttpPinotClientTransportFactory` looks for a property named `useMultiStageEngine` (uppercase "S") when building its connection properties.  The challenge is that most other code, examples and documentation refer to this property as `useMultistageEngine` (lowercase "S").  Since `getProperty` is case-sensitive, this can lead to unexpected behavior.  This property should be updated to reflect its case in other parts of the code.

### PR DESCRIPTION
**Pull request by timveil**
This PR standardizes the case of the property `useMultistageEngine` in `JsonAsyncHttpPinotClientTransportFactory`.  In all other places this property has a lowercase `s`.  In `JsonAsyncHttpPinotClientTransportFactory` the property was incorrectly referenced as `useMultiStageEngine` causing this property to be ignored when using the standard casing.

Fixes #11819

### CODE REVIEW

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/base/AbstractBaseConnection.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public void abort(Executor executor)
   @Override
   public void clearWarnings()
       throws SQLException {
-    throw new SQLFeatureNotSupportedException();
+    // no-op
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/base/AbstractBaseConnection.java:54]: This seems unrelated to this PR. Can you open a separate PR for these changes?

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/controller/PinotControllerTransport.java [hunk 2]
```diff
@@ -131,6 +131,7 @@ public ControllerTenantBrokerResponse getBrokersFromController(String controller
         _headers.forEach((k, v) -> requestBuilder.addHeader(k, v));
       }
 
+
```
Comment (Jackie-Jiang) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/controller/PinotControllerTransport.java:134]: (minor) extra empty line

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise

**Comment by Jackie-Jiang** [comment]
cc @xiangfu0 to also take a look

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/controller/PinotControllerTransport.java (hunk 2 above)
Comment (timveil) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/controller/PinotControllerTransport.java:134]: this is done

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/base/AbstractBaseConnection.java (hunk 1 above)
Comment (timveil) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/base/AbstractBaseConnection.java:54]: unfortunately these are part of the changes for #11813 that i based this PR on.

File: pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java [hunk 3]
```diff
@@ -109,7 +109,9 @@ public CompletableFuture<BrokerResponse> executeQueryAsync(String brokerAddress,
       json.put("sql", query);
       json.put("queryOptions", _extraOptionStr);
 
-      String url = String.format("%s://%s%s", _scheme, brokerAddress, _useMultiStageEngine ? "/query" : "/query/sql");
+      LOGGER.debug("Query will use Multistage Engine = {}", _useMultistageEngine);
```
Comment (xiangfu0) [inline pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java:112]: I suggest directly log the query url, which might be more useful.

**Review by xiangfu0 — APPROVED** [review]
lgtm

File: pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java (hunk 3 above)
Comment (xiangfu0) [reply pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransport.java:112]: ack

**Comment by xiangfu0** [comment]
Many thanks @timveil !
Congrats on the very first PR in Pinot!

**Pull request merged**